@xen-orchestra/xapi/src/vm.js

```diff
@@ -402,9 +402,24 @@ module.exports = class Vm {
     }
   }
 
+  @defer
   @cancelable
-  async snapshot($cancelToken, vmRef, nameLabel) {
+  async snapshot($cancelToken, $defer, vmRef, nameLabel) {
     const vm = await this.getRecord('VM', vmRef)
+    if (vm.power_state === 'Halted' && this._ignoreNobakVdis) {
```

Comment:
We can't unplug a VBD from a `Paused/Suspended` VM. It will trigger a `VM_BAD_POWER_STATE` error.

Reply:
Thanks for your test, can you add this in a comment?

```suggestion
    // cannot unplug VBDs on Running, Paused and Suspended VMs
    if (vm.power_state === 'Halted' && this._ignoreNobakVdis) {
```